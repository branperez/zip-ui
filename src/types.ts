export interface Location {
    postCode: string,
    country: string,
    countryAbv: string,
    places: [Place]
}

export interface Place {
    name: string,
    longitude: string,
    state: string,
    stateAbv: string,
    latitude: string
}

export interface Country {
    name: string,
    code: string 
}

export interface History {
    postCode: string,
    city: string,
    state: string,
    timeStamp: number 
}
