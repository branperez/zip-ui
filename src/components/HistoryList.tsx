import { Box, List, ListItem, ListItemText, Typography } from "@mui/material";
import React from "react";
import { History } from "../types";

const historyItems = ({ postCode, city, state, timeStamp }: History) => {
    const date = new Date(timeStamp * 1000)

    return (
        <ListItem key={timeStamp}>
            <ListItemText
                primary={`${city}, ${state}`}
                secondary={
                    <React.Fragment>
                        <Typography
                            component="span"
                            variant="body2"
                        >
                            {postCode} -- created at: 
                        </Typography>
                        {`  ${date.toLocaleString()}`}
                    </React.Fragment>
                }
            />
        </ListItem>
    )
}

const HistoryList = ({ previous }: { previous: History[] }) => {
    const historyList = previous.map(historyItems)

    return (
        <Box>
            <List>
                { historyList }
            </List>
        </Box>
    )
}

export default HistoryList