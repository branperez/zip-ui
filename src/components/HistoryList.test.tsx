import React from 'react';
import { render, screen } from '@testing-library/react';
import HistoryList from './HistoryList';
import { History } from '../types';
import renderer from 'react-test-renderer';

it('correctly renders History -- admittedly a pretty weeak test', () => {
    const item: History = { postCode: '70119', city: 'New Orleans', state: 'Louisiana', timeStamp: 0 }
    const history = [ item ]
    const component = renderer.create(
        <HistoryList previous={history} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
