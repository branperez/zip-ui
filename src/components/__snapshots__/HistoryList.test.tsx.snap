// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders History -- admittedly a pretty weeak test 1`] = `
<div
  className="MuiBox-root css-0"
>
  <ul
    className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      disabled={false}
    >
      <div
        className="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          New Orleans, Louisiana
        </span>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            70119
             -- created at:
          </span>
            12/31/1969, 7:00:00 PM
        </p>
      </div>
    </li>
  </ul>
</div>
`;
